### PR TITLE
chore: add tests for docker

### PR DESCRIPTION
**Pull request by amanda103**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,6 +59,7 @@ COPY requirements/local-inference.txt requirements-local-inference.txt
 
 RUN python3.8 -m pip install pip==${PIP_VERSION} \
   && pip install --no-cache -r requirements-base.txt \
+  && pip install --no-cache -r requirements-test.txt \
```
Comment (ryannikolaidis) [inline Dockerfile:62]: did adding this add much time to install and size to the image? probably easiest for now, but wondering as we try to slim this down what kind of impact this has.

File: Dockerfile [hunk 2]
```diff
@@ -60,5 +68,6 @@ RUN python3.8 -m pip install pip==${PIP_VERSION} \
   && pip install --no-cache "detectron2@git+https://github.com/facebookresearch/detectron2.git@v0.6#egg=detectron2"
 
 COPY unstructured unstructured
+COPY test_unstructured test_unstructured
```
Comment (ryannikolaidis) [inline Dockerfile:71]: we probably don't want to include these in the docker image right? maybe just a copy step before running the tests on the container in CI?

File: Dockerfile [hunk 3]
```diff
@@ -8,6 +8,12 @@ ARG UNSTRUCTURED
 RUN yum -y update && \
   yum -y install poppler-utils xz-devel which
 
+# Enable the EPEL repository
+RUN yum install -y epel-release && yum clean all
+
```
Comment (ryannikolaidis) [inline Dockerfile:13]: guessing centos dependency fun here? curios: was this needed for issues fetching pandoc? or something else?

File: Dockerfile (hunk 2 above)
Comment (awalker4) [reply Dockerfile:71]: I'd suggest mounting the test dir in during `make docker-test`. To ryan's point on the pip install, we could even get fancy and mount the local .venv to bring the test deps in at runtime. But yeah, certainly not a priority.

**Comment by awalker4** [comment]
LGTM! (I think the workflow file indentation is off. YAML is fun :) )

**Comment by ryannikolaidis** [comment]
> (I think the workflow file indentation is off. YAML is fun 😃 )

Good eye, hmmm, seems like we should be linting these... 🤨. Writing a ticket.

File: Dockerfile (hunk 3 above)
Comment (amanda103) [reply Dockerfile:13]: Yes this is a pandoc necessity, but the tests using pandoc are still complaining. Going to keep investigating!

File: Dockerfile (hunk 1 above)
Comment (amanda103) [reply Dockerfile:62]: Not a ton, but as Austin mentioned below maybe there's a better way to bring in test dependencies! Going to play around with that some more.

File: Dockerfile (hunk 3 above)
Comment (ryannikolaidis) [reply Dockerfile:13]: Argh! 😞 I definitely feel your pain there!

**Review by awalker4 — COMMENTED** [review]
LGTM! Just one question on the build step.

File: scripts/test.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+##!/bin/bash
+
+docker run --platform linux/amd64 -ti --rm \
+  -v ~work/unstructured/unstructured/test_unstructured/:/home/test_unstructured unstructured-dev:latest \
```
Comment (awalker4) [inline scripts/test.sh:4]: In order to be generic I think we'll want some form of $(PWD) here.

**Review by awalker4 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -128,3 +128,40 @@ jobs:
...
+        uses: mxschmitt/action-tmate@v3
+        with:
+          connection-method: 'https'
+      - name: Test Dockerfile
+        run: |
+          source .venv/bin/activate
+          make install
+          make install-detectron2
+          sudo apt-get update
+          sudo apt-get install -y libmagic-dev poppler-utils tesseract-ocr libreoffice pandoc
```
[the remark this answers is not in the document]
Comment (amanda103) [reply .github/workflows/ci.yml]: definitely not, good call!

**Pull request merged**